boot/freeldr/freeldr/disk/partition.c

```diff
@@ -17,20 +17,80 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-/*
- * TODO: This is here where we should add support for GPT partitions
- * as well as partitionless disks!
- */
-
 #ifndef _M_ARM
 #include <freeldr.h>
 
 #include <debug.h>
 
 DBG_DEFAULT_CHANNEL(DISK);
 
-/* This function serves to retrieve a partition entry for devices that handle partitions differently */
-DISK_GET_PARTITION_ENTRY DiskGetPartitionEntry = DiskGetMbrPartitionEntry;
+typedef enum
+{
+    DISKPART_UNKNOWN, /* Partitioning scheme not defined yet */
+    DISKPART_RAW,     /* Raw partitionless disk */
+    DISKPART_MBR,     /* MBR partitioning scheme */
```

Comment:
PARTITION_STYLE_MBR

Reply:
I see you're suggesting constants from MSDN, which are also used in kernel and usermode :slightly_smiling_face:

Reply:
And in (u)setup.